membership.go

```diff
@@ -354,20 +360,24 @@ func listenUDPMulticast(port int) error {
 		}
 
 		go func(addr *net.UDPAddr, bytes []byte) {
-			name, msgBytes := decodeMulticastAnnounceBytes(bytes)
-
-			if GetClusterName() == name {
-				msg, err := decodeMessage(addr.IP, msgBytes)
-				if err == nil {
-					logfTrace("Got multicast %v from %v code=%d\n",
-						msg.verb,
-						msg.sender.Address(),
-						msg.senderHeartbeat)
-
-					// Update statuses of the sender.
-					updateStatusesFromMessage(msg)
-				} else {
-					logError(err)
+			name, msgBytes, err := decodeMulticastAnnounceBytes(bytes)
+
+			if err != nil {
+				logDebug("Ignoring unexpected multicast message.")
```

Comment:
Should this (effectively) swallow the error? This could lead to messages being blackholed and lots of head-scratching. Maybe this should be higher-level error, and be more explicit about what was unexpected about the message?

Reply:
That is why I left the debug logging in there. When tracing for errors someone should notice this message but I guess you could argue that a higher level message would be appropriate.